source/intersperse.js

```diff
@@ -19,15 +19,17 @@ import _curry2 from './internal/_curry2.js';
  *
  *      R.intersperse('a', ['b', 'n', 'n', 's']); //=> ['b', 'a', 'n', 'a', 'n', 'a', 's']
  */
-var intersperse = _curry2(_checkForMethod('intersperse', function intersperse(separator, list) {
-  var out = [];
-  var idx = 0;
+var intersperse = _curry2(_checkForMethod('intersperse', function _intersperse(separator, list) {
   var length = list.length;
+  var out = Array(length * 2 - 1);
```

Comment:
This will throw an error if intersperse is passed an empty array. We can just bail early and return empty array in that case.

Reply:
Updated with a bail out when `length === 0`